File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
import re
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        s = s.strip()
        s = re.sub(r' {2,}', ' ', s)
        s = s.split(' ')
        s = s[::-1]
        return ' '.join(s)
            

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s)
    {
        string reversed = "", currWord = "";
        for (int i = 0; i < s.length(); i++)
        {
            if (s[i] == ' ' || i == s.length() - 1)
            {
                if (s[i] == ' ')
                {
                    for (; s[i] == ' '; i++);
                    i--;
                }
                if (i == s.length() - 1 && s[i] != ' ')
                {
                    currWord += s[i];
                }
                if (!currWord.empty())
                {
                    reversed = currWord + ' ' + reversed;
                    currWord = "";
                }
            }
            else
            {
                currWord += s[i];
            }
        }
        reversed.erase(reversed.length() - 1, 1);
        return reversed;
    }
};